context.go

```diff
@@ -153,7 +156,10 @@ func (s *Subscription) NextMsgWithContext(ctx context.Context) (*Msg, error) {
 	select {
 	case msg, ok = <-mch:
 		if !ok {
-			return nil, ErrConnectionClosed
+			s.mu.Lock()
```

Comment:
Since we repeat this four times, let's make it its own func.

Reply:
I see, will do.